components/web_discovery/browser/server_config_loader.cc

```diff
@@ -416,14 +459,33 @@ void ServerConfigLoader::OnPatternsGunzip(
     HandlePatternsStatus(false);
     return;
   }
-  auto parsed_patterns = ParsePatterns(*patterns_json);
+  background_task_runner_->PostTaskAndReplyWithResult(
+      FROM_HERE, base::BindOnce(&ParsePatterns, *patterns_json),
```

Comment:
make it `ParseAndWritePatterns` that will return parsed structure and a boolean indicating the write result, please do not blindly copy strings everywhere.